src/main/java/com/ldtteam/domumornamentum/block/AbstractBlockSlab.java

```diff
@@ -1,20 +1,175 @@
+    {
+        BlockPos blockpos = context.getClickedPos();
+        BlockState defaultstate = this.defaultBlockState();
+        BlockState blockstate = context.getLevel().getBlockState(blockpos);
+        FluidState fluidstate = context.getLevel().getFluidState(context.getClickedPos());
+        Direction direction = context.getClickedFace();
+
+
+        /** stacking, if clicked position lands RIGHT in middle on any axis
+         * also if clicked in any other side of side facing slabs*/
+        if (blockstate.is(this)) {
+            boolean xpos = context.getClickLocation().x - (double)context.getClickedPos().getX() == 0.5D;
+            boolean ypos = context.getClickLocation().y - (double)context.getClickedPos().getY() == 0.5D;
+            boolean zpos = context.getClickLocation().z - (double)context.getClickedPos().getZ() == 0.5D;
+            boolean xflag = context.getClickLocation().x - (double)context.getClickedPos().getX() > 0.5D;
+            boolean yflag = context.getClickLocation().y - (double)context.getClickedPos().getY() > 0.5D;
+            boolean zflag = context.getClickLocation().z - (double)context.getClickedPos().getZ() > 0.5D;
+            if (xpos || ypos || zpos){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+            if(blockstate.getValue(FACING) == Direction.UP && !yflag && direction != Direction.DOWN){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+            if(blockstate.getValue(FACING) == Direction.DOWN && yflag && direction != Direction.UP){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+            if(blockstate.getValue(FACING) == Direction.NORTH && !zflag && direction != Direction.SOUTH){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+            if(blockstate.getValue(FACING) == Direction.SOUTH && zflag && direction != Direction.NORTH){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+            if(blockstate.getValue(FACING) == Direction.EAST && xflag && direction != Direction.WEST){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+            if(blockstate.getValue(FACING) == Direction.WEST && !xflag && direction != Direction.EAST){
+                return blockstate.setValue(TYPE, SlabType.DOUBLE).setValue(WATERLOGGED, Boolean.valueOf(false));
+            }
+        }
+
+        /**if holding the shift key while clicking, set slab FACING (N/E/W/S)
+         * all facing slabs use BOTTOM TYPE
+         * else use the base class default */
+        if (context.isSecondaryUseActive()) {
```

Comment:
Didnt get what this does

Reply:
while sneak clicking, set facing to horizontal opposite, unless up or down then set up or down